### PR TITLE
En names

### PR DESCRIPTION
**Pull request by anand-nv**
# What does this PR do ?

Add support for additional alpha numeric sequences relevant to product names and expands whitelist

# Before your PR is "Ready for review"
**Pre checks**:
- [X] Did you write any new necessary tests?

**PR Type**:
- [X] New Feature

### CODE REVIEW

File: nemo_text_processing/inverse_text_normalization/en/taggers/date.py [hunk 1]
```diff
@@ -64,6 +64,15 @@ def _get_range_graph():
     return graph
 
 
+def _get_financial_period_graph():
+    h_ordinals = pynini.cross('first', '1') | pynini.cross('second', '2')
+    q_ordinals = h_ordinals | pynini.cross('third', '3') | pynini.cross('fourth', '4')
+
+    h_graph = h_ordinals + pynini.cross(' half', 'H') | pynini.cross(' h', 'H')
+    q_graph = q_ordinals + pynini.cross(' quarter', 'Q') | pynini.cross(' q', 'Q')
```
Comment (ekmb) [inline nemo_text_processing/inverse_text_normalization/en/taggers/date.py:72]: could you move these words to files, please?

File: nemo_text_processing/inverse_text_normalization/en/taggers/date.py [hunk 2]
```diff
@@ -93,11 +102,31 @@ def _get_thousands_graph():
     graph_ties = _get_ties_graph()
     graph_digits = _get_digits_graph()
     graph_thousands = _get_thousands_graph()
+    graph_ad_bc = (
+        (pynini.cross(" a d", ' AD') | pynini.cross(" b c", ' BC'))
+        | (pynini.cross(" anno domini", ' AD') | pynini.cross(" before christ", ' BC'))
+        | (pynini.cross(" c e", ' CE') | pynini.cross(" b c e", ' BCE'))
+        | (pynini.cross(" common era", ' CE') | pynini.cross(" before common era", ' BCE'))
+    )
```
Comment (ekmb) [inline nemo_text_processing/inverse_text_normalization/en/taggers/date.py:110]: could you move these words to files, please? when adding ITN for ASR+PC model it'll make integration easier

File: nemo_text_processing/inverse_text_normalization/en/taggers/date.py [hunk 3]
```diff
@@ -146,9 +175,22 @@ def __init__(self, ordinal: GraphFst):
             + month_graph
             + optional_graph_year
         )
+
+        """
+           period_fy = pynutil.insert("period: \"") + _get_financial_period_graph() + pynutil.insert("\"")
+               graph_fy = period_fy + insert_space + two_digit_year
```
Comment (ekmb) [inline nemo_text_processing/inverse_text_normalization/en/taggers/date.py:181]: do you need this comment?

File: nemo_text_processing/inverse_text_normalization/en/taggers/telephone.py [hunk 4]
```diff
@@ -28,12 +29,26 @@
 
 def get_serial_number(cardinal):
     """
-    any alphanumerical character sequence with at least one number with length greater equal to 3
+    any alphanumerical character sequence with at least one number with length greater equal to 3 and
+    excluding any numeric sequence preceded by 'a'
```
Comment (ekmb) [inline nemo_text_processing/inverse_text_normalization/en/taggers/telephone.py:33]: why excluding 'a'?

**Review by ekmb — CHANGES_REQUESTED** [review]
Could you this PR be merged after https://github.com/NVIDIA/NeMo-text-processing/pull/22 ?

File: nemo_text_processing/inverse_text_normalization/en/taggers/telephone.py (hunk 4 above)
Comment (anand-nv) [reply nemo_text_processing/inverse_text_normalization/en/taggers/telephone.py:33]: Updated the comment to read as follows " ... excluding any numeric sequence containing double digits (ties/teens) preceded by 'a'. This avoids cases like "a thirty six" being converted to "a36"  in "a thirty six times increase"

Comment (ekmb) [reply nemo_text_processing/inverse_text_normalization/en/taggers/telephone.py:33]: > Updated the comment to read as follows '_... excluding any numeric sequence containing double digits (ties/teens) preceded by 'a'. This avoids cases like "a thirty six" being converted to "a36" in "a thirty six times increase"_'

makes sense! thank you.

File: nemo_text_processing/inverse_text_normalization/en/taggers/date.py (hunk 1 above)
Comment (anand-nv) [reply nemo_text_processing/inverse_text_normalization/en/taggers/date.py]: Moved

File: nemo_text_processing/inverse_text_normalization/en/taggers/date.py (hunk 2 above)
Comment (anand-nv) [reply nemo_text_processing/inverse_text_normalization/en/taggers/date.py]: Moved to `year_suffix.tsv`.=

**Comment by anand-nv** [comment]
> Could you this PR be merged after #22 ?

Yes.

**Comment by yzhang123** [comment]
@anand-nv the #22 pr is merged, could you please fix the merge conflicts?

**Comment by anand-nv** [comment]
> @anand-nv the #22 pr is merged, could you please fix the merge conflicts?

Fixed

**Pull request merged**